hazelcast/src/main/java/com/hazelcast/map/impl/recordstore/AbstractRecordStore.java

```diff
@@ -208,4 +218,48 @@ public void disposeDeferredBlocks() {
     public Storage<Data, ? extends Record> getStorage() {
         return storage;
     }
+
+    @Override
+    public long getHits() {
+        return hits;
+    }
+
+    @Override
+    public long getLastAccessTime() {
+        return lastAccess;
+    }
+
+    @Override
+    public long getLastUpdateTime() {
+        return lastUpdate;
+    }
+
+    @Override
+    public void increaseHits() {
+        this.hits++;
+    }
+
+    @Override
+    public void setLastAccessTime(long time) {
+        this.lastAccess = Math.max(this.lastAccess, time);
+    }
+
+    @Override
+    public void setLastUpdateTime(long time) {
+        this.lastUpdate = Math.max(this.lastUpdate, time);
+    }
+
+
+    protected void updateStatsOnPut(boolean newRecord, long now) {
+        setLastUpdateTime(now);
+
+        if (!newRecord) {
+            updateStatsOnGet(now);
+        }
+    }
+
+    protected void updateStatsOnGet(long now) {
+        setLastAccessTime(now);
+        increaseHits();
```

Comment:
I'm not sure about that. We're trying to reach the total hit count of a map, right? Imagine that a map is cleared, then this means the hit count is 0 now? This should not be the case, IMO.